msgsecret.go

```diff
@@ -115,6 +118,38 @@ func (cli *Client) encryptMsgSecret(chat, origSender types.JID, origMsgID types.
 	return ciphertext, iv, nil
 }
 
+func (cli *Client) applyBotMessageHKDF(messageSecret []byte) []byte {
+	return hkdfutil.SHA256(messageSecret, nil, []byte(EncSecretBotMsg), 32)
+}
+
+func (cli *Client) createIncomingBotMessageSecret(messageId types.MessageID, targetJid types.JID, botJid types.JID, messageSecret []byte) []byte {
+	targetJidStr := targetJid.ToNonAD().String()
+	botJidStr := botJid.ToNonAD().String()
+
+	hkdfSecret := make([]byte, 0, len(messageId)+len(targetJidStr)+len(botJidStr))
+	hkdfSecret = append(hkdfSecret, messageId...)
+	hkdfSecret = append(hkdfSecret, targetJidStr...)
+	hkdfSecret = append(hkdfSecret, botJidStr...)
+
+	secretKey := hkdfutil.SHA256(cli.applyBotMessageHKDF(messageSecret), nil, hkdfSecret, 32)
+
+	return secretKey
+}
+
+func (cli *Client) decryptBotMessage(messageSecret []byte, msMsg messageEncryptedSecret, messageId types.MessageID, targetSenderJid types.JID, info *types.MessageInfo) ([]byte, error) {
+	// gcm decrypt key generation
+	newKey := cli.createIncomingBotMessageSecret(messageId, targetSenderJid, info.Sender, messageSecret)
+
+	additionalData := []byte(fmt.Sprintf("%s\x00%s", messageId, info.Sender))
```

Comment:
I think this is equivalent and makes the `createIncomingBotMessageSecret` function redundant?

```suggestion
	newKey, additionalData := cli.generateMsgSecretKey("", info.Sender, messageId, targetSenderJid, applyBotMessageHKDF(messageSecret))
```

Reply:
I'll test it

Reply:
I had considered this before, waited till u see the whole picture before I asked about it